src/amo/components/RatingManager/index.js

```diff
@@ -182,40 +190,91 @@ export class RatingManagerBase extends React.Component<InternalProps, State> {
     }
   };
 
-  render() {
-    const { i18n, addon, userId, userReview, version } = this.props;
-    const { showTextEntry } = this.state;
-    const isLoggedIn = Boolean(userId);
+  cancelTextEntry = (event: SyntheticEvent<any>) => {
+    event.preventDefault();
+    this.setState({ showTextEntry: false });
+  };
 
-    invariant(addon, 'addon is required');
-    invariant(version, 'version is required');
+  renderTextEntry() {
+    const { _config, userReview } = this.props;
+
+    if (!userReview || !this.isSignedIn()) {
```

Comment:
There's a test for not rendering the `AddonReviewManager` if the user isn't signed in, but not for the case where `userReview` is falsey.

Reply:
Ah, that's true. That state actually wasn't rendering correctly either. I hope to revisit these render states when I can get rid of the old behavior in RatingManager. It should be controlled more from Redux.